crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -171,19 +171,40 @@ where
                 },
             );
 
-        // TODO: Use multi get objects
         let rpc = self.rpc_client.clone();
-        let all_mutated_objects = join_all(all_mutated.into_iter().map(|(id, version, status)| {
-            rpc.read_api()
-                .try_get_parsed_past_object(id, version, SuiObjectDataOptions::bcs_lossless())
-                .map(move |resp| (resp, status))
-        }))
-        .await
-        .into_iter()
-        .try_fold(vec![], |mut acc, (response, status)| {
-            acc.push((status, response?.into_object()?));
-            Ok::<_, Error>(acc)
-        })?;
+        let all_mutated_objects =
+            join_all(all_mutated.chunks(MULTI_GET_CHUNK_SIZE).map(|objects| {
+                let wanted_past_object_statuses: Vec<ObjectStatus> =
+                    objects.iter().map(|(_, _, status)| *status).collect();
+
+                let wanted_past_object_request = objects
+                    .iter()
+                    .map(|(id, seq_num, _)| SuiGetPastObjectRequest {
+                        object_id: *id,
+                        version: *seq_num,
+                    })
+                    .collect();
+
+                rpc.read_api()
+                    .try_multi_get_parsed_past_object(
+                        wanted_past_object_request,
+                        SuiObjectDataOptions::bcs_lossless(),
+                    )
+                    .map(move |resp| (resp, wanted_past_object_statuses))
+            }))
+            .await
+            .into_iter()
+            .try_fold(vec![], |mut acc, chunk| {
+                let object_datas: Vec<SuiObjectData> = chunk
+                    .0?
+                    .into_iter()
+                    // Unwrap should be ok here as the batched response will fail if any objects are invalid.
+                    .map(|resp| resp.into_object().unwrap())
```

Comment:
I would not use `unwrap` here since it's possible that the batch response is successful but the conversion to object is not